### ISSUE TITLE
FR: Indicate total page number for each book in exported JSON

### ISSUE THREAD

**Issue by 1over137** [koreader/koreader#8674]
**Does your feature request involve difficulty completing a task? Please describe.**
I'm developing a tool to quickly generate flashcards from reading highlights for language learning. The intended use is that users highlight unknown words, and afterwards export the highlights. Then, the tool should be able to backreference the words in the book to find context. However, in order to be reasonably accurate, there is a need to know the approximate location of the highlight in the book. The problem is that the page number in the clippings file is dependent on the font size and there is no simple way to determine how many pages the book is on the user's device.

**Describe the solution you'd like**
Include a field in the JSON export for the total number of pages for each book listed.

**Describe alternatives you've considered**
Have the user input the total pages at current settings, but that is cumbersome if there are many books.

**Comment by hius07** [comment]
Can you give more details about your tool.
As you noted, number of pages depends on font/margins/line spacing etc.
You may need other info to locate the position in a book.

**Comment by 1over137** [comment]
> Can you give more details about your tool. As you noted, number of pages depends on font/margins/line spacing etc. You may need other info to locate the position in a book.

https://github.com/FreeLanguageTools/ssmtool
It's not in there yet.
The idea is that the user will simply highlight words and the tool will both look the words up in a dictionary and also find the context they appeared in in the book. I know knowing the total number of pages isn't very precise, but it should work for most cases since it would significantly narrow the range of the search where there would only be a single instance of the named word. Of course, it would also be nice to have more precise options such has by having highlights directly exported with their own sentences, but these will probably require more work.

**Comment by hius07** [comment]
> find the context they appeared in in the book

How to do that outside of KOReader? Seems not easy for your tool.

**Comment by 1over137** [comment]
> > find the context they appeared in in the book
> 
> How to do that outside of KOReader? Seems not easy for your tool.

My plan is to just scan the book for all instance of the word and pick the one closest to the location indicated by the percentage of the location to the total length. From a few experiment with kindle on a similar idea it works reasonably well. Though, a nicer addition would be to simply list the index of the word from the beginning, like 29481 or use a percentage for words rather than pages to be more precise.

**Comment by poire-z** [comment]
Pinging @uroybd in case that's something that could be added to your newly exporter stuff.
If it's not relevant (because cre total page numbers is volatile), let's close this issue.

**Comment by uroybd** [comment]
I can add the total page number. Again, it depends on the font size, right?

**Comment by poire-z** [comment]
With cre yes (and may change with mostly any other setting than font size).
But with PDF, it is fixed - so, this might help in some cases.

**Comment by uroybd** [comment]
Well... No harm adding this. Will try.

**Comment by 1over137** [comment]
Is this fixed for non-PDF formats?
I ended up simply parsing the metadata.lua files to avoid having to ask the user to export every time. This might still come in handy for future use though. Thanks for the work.

**Comment by uroybd** [comment]
> Is this fixed for non-PDF formats? I understand they are volatile, but it should be constant within the same export.
> I ended up simply parsing the metadata.lua files to avoid having to ask the user to export every time. This might still come in handy for future use though. Thanks for the work.

That is kind of out of scope for exporter plugin. See #9020. Once it is done, it will provide a consistent logical page count to be used everywhere.
